lib/pure/parsexml.nim

```diff
@@ -299,15 +299,15 @@ template piRest*(my: XmlParser): string =
   assert(my.kind == xmlPI)
   my.b
 
-proc rawData*(my: var XmlParser): string {.inline.} =
+proc rawData*(my: sink XmlParser): string {.inline.} =
   ## returns the underlying 'data' string by reference.
```

Comment:
Since the file is already being modified, perhaps the documentation should be updated to state *why* this is unsafe as well?

Reply:
With this change its actually safe with `--gc:arc`.
But in general its unsafe because it returns it by reference/shallow copy.